File: GlobalTax-FE/src/app/shared/models/responseDto.ts
interface TaxBreakdown {
  range: string;
  rate: number;
  taxedAmount: number;
  taxOwed: number;
}

interface TaxResult {
  totalTax: number;
  effectiveRate: number;
  breakdown: TaxBreakdown[];
}
 
interface TaxBracketDetail {
  // id: number;
  level: number;
  limit: number;
  rate: number;
  // countryCode: string;
}

interface ValidationResponseDto {
  success: boolean;
  response: string[];
}

interface UpdateBracketResponseDto {
  success: boolean;
  response: string;
}

interface GetTaxBracketResponseDto {
  success: boolean;
  response: string;
  brackets: TaxBracketDetail[];
}


//

interface Country {
  // id: number;
  countryCode: string;
  countryName: string;
}


interface GetCountryResponseDto {
  success: boolean;
  country: Country[];
}

export type { 
  TaxBreakdown, 
  TaxResult, 
  TaxBracketDetail,
  ValidationResponseDto,
  UpdateBracketResponseDto,
  GetTaxBracketResponseDto,
  GetCountryResponseDto,
  Country
};

File: GlobalTax-BE/src/Application/Models/ResponseDto.cs
using GlobalTax.Domain;

namespace GlobalTax.Application.Models;

// Tax
public record ValidationResponseDto(bool? Success, List<string> Response);

public record UpdateBracketResponseDto(bool? Success, string Response);

public record GetTaxBracketResponseDto(bool? Success, string Response, List<TaxBracketDetail>? Brackets);


// Country
public record GetCountryResponseDto(bool? Success, string Response, List<Country>? Country);

